### ISSUE TITLE
Issue with LDFLAGS - Trying to install Ruby 2.3.4 on MacOS Mojave 10.14.5

### ISSUE THREAD

**Issue by jonathonpday** [rvm/rvm#4862]
I Am trying to install Ruby 2.2.4 on my machine using:

`rvm install ruby-2.2.4`

However it keeps failing at configuration:

`ruby-2.2.4 - #applying patch /Users/jonathonday/.rvm/patches/ruby/2.2.4/openssl3.patch.`
`ruby-2.2.4 - #configuring......`
`Error running './configure --prefix=/Users/jonathonday/.rvm/rubies/ruby-2.2.4 --with-opt-dir=/usr/local/opt/libyaml:/usr/local/opt/libksba:/usr/local/opt/readline:/usr/local/opt/zlib:/usr/local/opt/openssl@1.1 --disable-install-doc --enable-shared',`
`please read /Users/jonathonday/.rvm/log/1579711928_ruby-2.2.4/configure.log`
`There has been an error while running configure. Halting the installation.`

When I open the log I get the following in the last 10 lines:
`checking for memory.h... yes`
 `checking for strings.h... yes`
 `checking for inttypes.h... yes`
 `checking for stdint.h... yes`
 `checking for unistd.h... yes`
 `checking minix/config.h usability... no`
 `checking minix/config.h presence... no`
 `checking for minix/config.h... no`
 `checking whether it is safe to define __EXTENSIONS__... yes`
 `checking for cd using physical directory... cd -P`
 `checking whether CFLAGS is valid... yes`
 `checking whether LDFLAGS is valid... no`
 `configure: error: something wrong with LDFLAGS="-L/Users/jonathonday/.rvm/usr/lib -L/opt/local/lib"` 

Can anyone tell me how best to proceed with this, I have tried following previous answers and updating xcode etc but I have had no luck 

I  also don't know where the flags are and how to amend them?

**Comment by rutsoluo** [comment]
Ruby 2.3.4 won't work with openssl 1.1, it will only work with openssl 1.0.x
This is what I did:
unset any FLAGS suggested by homebrew install openssl@1.1, CFLAGS, LDFLAGS, etc...
rvm install openssl # this will install openssl 1.0.2
rvm install 2.3.4 --with-openssl-dir=$HOME/.rvm/usr

**Comment by jonathonpday** [comment]
> Ruby 2.3.4 won't work with openssl 1.1, it will only work with openssl 1.0.x
> This is what I did:
> If you installed openssl via home brew, unset the following flags:
> unset CFLAGS
> unset LDFLAGS
> unset CPPFLAGS
> rvm install openssl # this will install openssl 1.0.2
> rvm install 2.3.4 --with-openssl-dir=$HOME/.rvm/usr #this will recompile ruby with openssl 1.0.2

Thank you I will absolutely give this a go. Just to ask as well (I am quite new to Ruby) where is this sort of information about gem compatibility with Ruby versions stored? IS there anywhere particularly I can look to find this out myself?

Cheers again!

**Comment by jonathonpday** [comment]
Unfortunately This did not work I unset all the flags as instructed and installed openssl 1.0.2:

`Warning: openssl 1.0.2t is already installed and up-to-date
To reinstall 1.0.2t, run brew reinstall openssl`
`

However, after running `rvm install ruby-2.3.4` again I still get the same issue:

`Error running './configure --prefix=/Users/jonathonday/.rvm/rubies/ruby-2.3.4 --with-opt-dir=/usr/local/opt/libyaml:/usr/local/opt/libksba:/usr/local/opt/readline:/usr/local/opt/zlib:/usr/local/opt/openssl@1.1 --disable-install-doc --enable-shared',
please read /Users/jonathonday/.rvm/log/1580171541_ruby-2.3.4/configure.log
There has been an error while running configure. Halting the installation.`

Is there anything else I can try to do?

**Comment by rutsoluo** [comment]
you should install ruby by running the following command, as you see in the error message, it is still trying to grab openssl from `/usr/local/opt/openssl@1.1`

```
rvm install 2.3.4 --with-openssl-dir=$HOME/.rvm/usr
```

**Comment by jonathonpday** [comment]
Unfortunately still no joy!

```
jonathonday@Jonathons-MacBook-Air:~/ruby$ rvm remove 2.3.4
ruby-2.3.4 - #already gone
Now using system ruby.
jonathonday@Jonathons-MacBook-Air:~/ruby$ rvm install 2.3.4 --with-openssl-dir=$HOME/.rvm/usr
Warning, new version of rvm available '1.29.9', you are using older version '1.29.9-next'.
You can disable this warning with:   echo rvm_autoupdate_flag=0 >> ~/.rvmrc
You can enable auto-update with:     echo rvm_autoupdate_flag=2 >> ~/.rvmrc
You can update manually with:        rvm get VERSION                         (e.g. 'rvm get stable')

Checking requirements for osx.
Certificates bundle '/usr/local/etc/openssl@1.1/cert.pem' is already up to date.
Requirements installation successful.
Installing Ruby from source to: /Users/jonathonday/.rvm/rubies/ruby-2.3.4, this may take a while depending on your cpu(s)...
ruby-2.3.4 - #downloading ruby-2.3.4, this may take a while depending on your connection...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 13.7M  100 13.7M    0     0  2473k      0  0:00:05  0:00:05 --:--:-- 3371k
ruby-2.3.4 - #extracting ruby-2.3.4 to /Users/jonathonday/.rvm/src/ruby-2.3.4.....
ruby-2.3.4 - #applying patch /Users/jonathonday/.rvm/patches/ruby/ruby_2_3_gcc7.patch.
ruby-2.3.4 - #applying patch /Users/jonathonday/.rvm/patches/ruby/2.3.4/fix_rb_thread_fd_close.patch.
ruby-2.3.4 - #applying patch /Users/jonathonday/.rvm/patches/ruby/2.3.4/random_c_using_NR_prefix.patch.
ruby-2.3.4 - #configuring......
Error running './configure --prefix=/Users/jonathonday/.rvm/rubies/ruby-2.3.4 --with-openssl-dir=/Users/jonathonday/.rvm/usr --with-opt-dir=/usr/local/opt/libyaml:/usr/local/opt/libksba:/usr/local/opt/readline:/usr/local/opt/zlib:/usr/local/opt/openssl@1.1 --disable-install-doc --enable-shared',
please read /Users/jonathonday/.rvm/log/1580173841_ruby-2.3.4/configure.log
There has been an error while running configure. Halting the installation.
```

**Comment by rutsoluo** [comment]
how about?

```
rvm pkg remove openssl
rvm remove 2.3.4
rvm install 2.3.4 --with-openssl-dir=$HOME/.rvm/usr
```

**Comment by jonathonpday** [comment]
Still the same result - followed these instructions in order twice to be sure!

```
jonathonday@Jonathons-MacBook-Air:~/ruby$ rvm remove 2.3.4
ruby-2.3.4 - #removing src/ruby-2.3.4..
Now using system ruby.
jonathonday@Jonathons-MacBook-Air:~/ruby$ rvm install 2.3.4 --with-openssl-dir=$HOME/.rvm/usr
Warning, new version of rvm available '1.29.9', you are using older version '1.29.9-next'.
You can disable this warning with:   echo rvm_autoupdate_flag=0 >> ~/.rvmrc
You can enable auto-update with:     echo rvm_autoupdate_flag=2 >> ~/.rvmrc
You can update manually with:        rvm get VERSION                         (e.g. 'rvm get stable')

Checking requirements for osx.
Certificates bundle '/usr/local/etc/openssl@1.1/cert.pem' is already up to date.
Requirements installation successful.
Installing Ruby from source to: /Users/jonathonday/.rvm/rubies/ruby-2.3.4, this may take a while depending on your cpu(s)...
ruby-2.3.4 - #downloading ruby-2.3.4, this may take a while depending on your connection...
ruby-2.3.4 - #extracting ruby-2.3.4 to /Users/jonathonday/.rvm/src/ruby-2.3.4.....
ruby-2.3.4 - #applying patch /Users/jonathonday/.rvm/patches/ruby/ruby_2_3_gcc7.patch.
ruby-2.3.4 - #applying patch /Users/jonathonday/.rvm/patches/ruby/2.3.4/fix_rb_thread_fd_close.patch.
ruby-2.3.4 - #applying patch /Users/jonathonday/.rvm/patches/ruby/2.3.4/random_c_using_NR_prefix.patch.
ruby-2.3.4 - #configuring......
Error running './configure --prefix=/Users/jonathonday/.rvm/rubies/ruby-2.3.4 --with-openssl-dir=/Users/jonathonday/.rvm/usr --with-opt-dir=/usr/local/opt/libyaml:/usr/local/opt/libksba:/usr/local/opt/readline:/usr/local/opt/zlib:/usr/local/opt/openssl@1.1 --disable-install-doc --enable-shared',
please read /Users/jonathonday/.rvm/log/1580198109_ruby-2.3.4/configure.log
There has been an error while running configure. Halting the installation.
```

**Comment by rutsoluo** [comment]
export SSL_CERT_DIR=$HOME/.rvm/usr/ssl
export SSL_CERT_FILE=$HOME/.rvm/usr/ssl/cert.pem

rvm remove 2.3.4
rvm install 2.3.4 --with-openssl-dir=$HOME/.rvm/usr -with-openssl_include=$HOME/.rvm/usr/include

**Comment by jonathonpday** [comment]
Got the following:

`
nathonday@Jonathons-MacBook-Air:~/ruby$ export SSL_CERT_DIR=$HOME/.rvm/usr/ssl
jonathonday@Jonathons-MacBook-Air:~/ruby$  export SSL_CERT_FILE=$HOME/.rvm/usr/ssl/cert.pem
jonathonday@Jonathons-MacBook-Air:~/ruby$ rvm remove 2.3.4
ruby-2.3.4 - #removing src/ruby-2.3.4..
Now using system ruby.
jonathonday@Jonathons-MacBook-Air:~/ruby$  rvm install 2.3.4 --with-openssl-dir=$HOME/.rvm/usr -with-openssl_include=$HOME/.rvm/usr/include
Warning, new version of rvm available '1.29.9', you are using older version '1.29.9-next'.
You can disable this warning with:   echo rvm_autoupdate_flag=0 >> ~/.rvmrc
You can enable auto-update with:     echo rvm_autoupdate_flag=2 >> ~/.rvmrc
You can update manually with:        rvm get VERSION                         (e.g. 'rvm get stable')

Unrecognized command line flag: '-with-openssl_include=/Users/jonathonday/.rvm/usr/include'
Run `rvm help` to see usage information

`

**Comment by jonathonpday** [comment]
Thanks a lot for your help with this, can i just try ask what we are trying to do?

**Comment by rutsoluo** [comment]
My bad should be --with-openssl_include
export SSL_CERT_DIR=$HOME/.rvm/usr/ssl
export SSL_CERT_FILE=$HOME/.rvm/usr/ssl/cert.pem

rvm remove 2.3.4
rvm install 2.3.4 --with-openssl-dir=$HOME/.rvm/usr --with-openssl_include=$HOME/.rvm/usr/include

**Comment by rutsoluo** [comment]
As you can in the previous error message, it is trying to grab the ssl certificate from openssl 1.1 which is not what we want. We want it to grab the openssl certificate from openssl 1.0.2 which rvm installed.

**Comment by jonathonpday** [comment]
The same thing:

```
jonathonday@Jonathons-MacBook-Air:~$ export SSL_CERT_DIR=$HOME/.rvm/usr/ssl
jonathonday@Jonathons-MacBook-Air:~$ export SSL_CERT_FILE=$HOME/.rvm/usr/ssl/cert.pem
jonathonday@Jonathons-MacBook-Air:~$ rvm remove 2.3.4
ruby-2.3.4 - #already gone
Now using system ruby.
jonathonday@Jonathons-MacBook-Air:~$ rvm remove 2.3.4
ruby-2.3.4 - #already gone
Now using system ruby.
jonathonday@Jonathons-MacBook-Air:~$ rvm install 2.3.4 --with-openssl-dir=$HOME/.rvm/usr --with-openssl_include=$HOME/.rvm/usr/include
Warning, new version of rvm available '1.29.9', you are using older version '1.29.9-next'.
You can disable this warning with:   echo rvm_autoupdate_flag=0 >> ~/.rvmrc
You can enable auto-update with:     echo rvm_autoupdate_flag=2 >> ~/.rvmrc
You can update manually with:        rvm get VERSION                         (e.g. 'rvm get stable')

Checking requirements for osx.
Certificates bundle '/usr/local/etc/openssl@1.1/cert.pem' is already up to date.
Requirements installation successful.
Installing Ruby from source to: /Users/jonathonday/.rvm/rubies/ruby-2.3.4, this may take a while depending on your cpu(s)...
ruby-2.3.4 - #downloading ruby-2.3.4, this may take a while depending on your connection...
ruby-2.3.4 - #extracting ruby-2.3.4 to /Users/jonathonday/.rvm/src/ruby-2.3.4.....
ruby-2.3.4 - #applying patch /Users/jonathonday/.rvm/patches/ruby/ruby_2_3_gcc7.patch.
ruby-2.3.4 - #applying patch /Users/jonathonday/.rvm/patches/ruby/2.3.4/fix_rb_thread_fd_close.patch.
ruby-2.3.4 - #applying patch /Users/jonathonday/.rvm/patches/ruby/2.3.4/random_c_using_NR_prefix.patch.
ruby-2.3.4 - #configuring......
Error running './configure --prefix=/Users/jonathonday/.rvm/rubies/ruby-2.3.4 --with-openssl-dir=/Users/jonathonday/.rvm/usr --with-openssl_include=/Users/jonathonday/.rvm/usr/include --with-opt-dir=/usr/local/opt/libyaml:/usr/local/opt/libksba:/usr/local/opt/readline:/usr/local/opt/zlib:/usr/local/opt/openssl@1.1 --disable-install-doc --enable-shared',
please read /Users/jonathonday/.rvm/log/1580231470_ruby-2.3.4/configure.log
There has been an error while running configure. Halting the installation.
```

**Comment by jonathonpday** [comment]
Is there a resource or docs you can recommend to learn about this stuff? Always been a bugbear of mine with Ruby - things always seem to break with gems and ruby versions etc?

**Comment by rutsoluo** [comment]
I just went through the same hell installing ruby 2.3.3, had to scour through the internet to find answers.

**Comment by jonathonpday** [comment]
Fair enough - is there anything else we can try?

**Comment by rutsoluo** [comment]
```
brew uninstall --ignore-dependencies openssl@1.1
export SSL_CERT_DIR=$HOME/.rvm/usr/ssl
export SSL_CERT_FILE=$HOME/.rvm/usr/ssl/cert.pem
rvm pkg remove openssl
rvm pkg install openssl
rvm remove 2.3.4
rvm install 2.3.4 --with-openssl-dir=$HOME/.rvm/usr --with-openssl-include=$HOME/.rvm/usr/include --with-openssl-lib=$HOME/.rvm/usr/lib
brew install openssl1.1
```

**Comment by jonathonpday** [comment]
Unfortunately still the same:

```
jonathonday@Jonathons-MacBook-Air:~$ rvm install 2.3.4 --with-openssl-dir=$HOME/.rvm/usr --with-openssl-include=$HOME/.rvm/usr/include --with-openssl-lib=$HOME/.rvm/usr/lib
Warning, new version of rvm available '1.29.9', you are using older version '1.29.9-next'.
You can disable this warning with:   echo rvm_autoupdate_flag=0 >> ~/.rvmrc
You can enable auto-update with:     echo rvm_autoupdate_flag=2 >> ~/.rvmrc
You can update manually with:        rvm get VERSION                         (e.g. 'rvm get stable')

Checking requirements for osx.
Certificates bundle '/usr/local/etc/openssl@1.1/cert.pem' is already up to date.
Requirements installation successful.
Installing Ruby from source to: /Users/jonathonday/.rvm/rubies/ruby-2.3.4, this may take a while depending on your cpu(s)...
ruby-2.3.4 - #downloading ruby-2.3.4, this may take a while depending on your connection...
ruby-2.3.4 - #extracting ruby-2.3.4 to /Users/jonathonday/.rvm/src/ruby-2.3.4.....
ruby-2.3.4 - #applying patch /Users/jonathonday/.rvm/patches/ruby/ruby_2_3_gcc7.patch.
ruby-2.3.4 - #applying patch /Users/jonathonday/.rvm/patches/ruby/2.3.4/fix_rb_thread_fd_close.patch.
ruby-2.3.4 - #applying patch /Users/jonathonday/.rvm/patches/ruby/2.3.4/random_c_using_NR_prefix.patch.
ruby-2.3.4 - #configuring......
Error running './configure --prefix=/Users/jonathonday/.rvm/rubies/ruby-2.3.4 --with-openssl-dir=/Users/jonathonday/.rvm/usr --with-openssl-include=/Users/jonathonday/.rvm/usr/include --with-openssl-lib=/Users/jonathonday/.rvm/usr/lib --with-opt-dir=/usr/local/opt/libyaml:/usr/local/opt/libksba:/usr/local/opt/readline:/usr/local/opt/zlib:/usr/local/opt/openssl@1.1 --disable-install-doc --enable-shared',
please read /Users/jonathonday/.rvm/log/1580247637_ruby-2.3.4/configure.log
There has been an error while running configure. Halting the installation.
```

**Comment by rutsoluo** [comment]
someone said this work for them:

```
rvm install 2.3.4 --disable-binary
```

**Comment by jonathonpday** [comment]
I think I got it to work.

I did some googling (Cheers!) and basically ran `rvm implode` and scrubbed all rvm references from my system and started again.

It now finally seems to have installed after a fashion:

```
ruby-2.3.4 - #generating global wrappers.................
Error running 'run_gem_wrappers regenerate',
please read /Users/jonathonday/.rvm/log/1580252158_ruby-2.3.4/gemset.wrappers.global.log
ruby-2.3.4 - #gemset created /Users/jonathonday/.rvm/gems/ruby-2.3.4
ruby-2.3.4 - #importing gemsetfile /Users/jonathonday/.rvm/gemsets/default.gems evaluated to empty gem list
ruby-2.3.4 - #generating default wrappers.................
Error running 'run_gem_wrappers regenerate',
please read /Users/jonathonday/.rvm/log/1580252158_ruby-2.3.4/gemset.wrappers.default.log
ruby-2.3.4 - #adjusting #shebangs for (gem irb erb ri rdoc testrb rake).
Install of ruby-2.3.4 - #complete 
Please be aware that you just installed a ruby that requires        3 patches just to be compiled on an up to date linux system.
This may have known and unaccounted for security vulnerabilities.
Please consider upgrading to ruby-2.7.0 which will have all of the latest security patches.
Ruby was built without documentation, to build it run: rvm docs generate-ri
```

Thank you so much for your help and patience with this I do very much appreciate it.

**Comment by rutsoluo** [comment]
Congrats!

I was actually going to suggest that next but since you did it and got it to work.

**Comment by suryadheeshjith** [comment]
Hate to be that guy. But rvm implode didn't work for me...... I tried everything suggested here but nothing seems to work for me.


```
suryad@x86_64-apple-darwin13:~ rvm install 2.7
Searching for binary rubies, this might take some time.
No binary rubies available for: osx/10.15/x86_64/ruby-2.7.0.
Continuing with compilation. Please read 'rvm help mount' to get more information on binary rubies.
Checking requirements for osx.
We don't know how to ensure that selected compiler 'x86_64-apple-darwin13.4.0-clang' exists.
Certificates bundle '/usr/local/etc/openssl@1.1/cert.pem' is already up to date.
Requirements installation successful.
Warning: found user selected compiler 'x86_64-apple-darwin13.4.0-clang', this will suppress RVM auto detection mechanisms.
Installing Ruby from source to: /Users/suryad/.rvm/rubies/ruby-2.7.0, this may take a while depending on your cpu(s)...
ruby-2.7.0 - #downloading ruby-2.7.0, this may take a while depending on your connection...

ruby-2.7.0 - #extracting ruby-2.7.0 to /Users/suryad/.rvm/src/ruby-2.7.0 - please wait
ruby-2.7.0 - #configuring - please wait
Error running './configure --prefix=/Users/suryad/.rvm/rubies/ruby-2.7.0  --with-opt-dir=/usr/local/opt/libyaml:/usr/local/opt/libksba:/usr/local/opt/readline:/usr/local/opt/zlib:/usr/local/opt/openssl@1.1 --disable-install-doc --enable-shared',
please read /Users/suryad/.rvm/log/1586458749_ruby-2.7.0/configure.log
There has been an error while running configure. Halting the installation.
```


Then I tried this 
```
unset CFLAGS
unset LDFLAGS
unset CPPFLAGS
(base)  suryad@x86_64-apple-darwin13 ~ rvm install openssl
Unrecognized command line argument: openssl
```
